File: elixir/day1/lib/part1.ex
defmodule Part1 do
  def filter_digits(char), do: String.match?(char, ~r/\d/)

  def calculate_sum(numbers) do
    String.to_integer(Enum.at(numbers, 0)) * 10 + String.to_integer(Enum.at(numbers, -1))
  end

  def process_line(line) do
    line
    |> String.graphemes()
    |> Enum.filter(&filter_digits/1)
  end

  def part1(input) do
    input
    |> String.split("\n", trim: true)
    |> Enum.map(&process_line/1)
    |> Enum.map(&calculate_sum/1)
    |> Enum.sum()
  end
end

File.read!(~c"input.txt")
|> Part1.part1()
|> IO.inspect(label: "part1")

File: ruby/day1/part1.rb
# frozen_string_literal: true

def part1(input)
  input.split("\n")
       .filter_map { |line| line.chars.filter_map { |char| char if char.match?(/\d/) } }
       .sum { |numbers| (numbers.first.to_i * 10) + numbers.last.to_i }
end

if __FILE__ == $PROGRAM_NAME
  input = File.read('input.txt')
  p part1(input)
end
